extensions/frontend-tiles/src/FrontendTiles.ts

```diff
@@ -3,23 +3,185 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 
-import { IModelConnection, SpatialTileTreeReferences, SpatialViewState } from "@itwin/core-frontend";
+import { AccessToken, Logger } from "@itwin/core-bentley";
+import { IModelApp, IModelConnection, SpatialTileTreeReferences, SpatialViewState } from "@itwin/core-frontend";
+import { loggerCategory } from "./LoggerCategory";
 import { createBatchedSpatialTileTreeReferences } from "./BatchedSpatialTileTreeRefs";
 
 /** A function that can provide the base URL where a tileset representing all of the spatial models in a given iModel are stored.
  * The tileset is expected to reside at "baseUrl/tileset.json" and to have been produced by the [mesh export service](https://developer.bentley.com/apis/mesh-export/).
  * If no such tileset exists for the given iModel, return `undefined`.
  * @see [[FrontendTilesOptions.computeSpatialTilesetBaseUrl]].
- * @alpha
+ * @beta
  */
 export type ComputeSpatialTilesetBaseUrl = (iModel: IModelConnection) => Promise<URL | undefined>;
 
+function createMeshExportServiceQueryUrl(args: { iModelId: string, urlPrefix?: string, changesetId?: string }): string {
+  const prefix = args.urlPrefix ?? "";
+  let url = `https://${prefix}api.bentley.com/mesh-export/?iModelId=${args.iModelId}&$orderBy=date:desc`;
+  if (args.changesetId)
+    url = `${url}&changesetId=${args.changesetId}`;
+
+  return url;
+}
+
+/** Represents the result of a [mesh export](https://developer.bentley.com/apis/mesh-export/operations/get-export/#export).
+ * @see [[queryCompletedMeshExports]].
+ * @beta
```

Comment:
do we really want to export this? shouldnt this be internal? this value is derived from what gets returned by apim; who knows what the repsonse will look like in a yr, maybe theyll drop `_links`

Reply:
It's not just derived from, it mirrors the [API](https://developer.bentley.com/apis/mesh-export/operations/get-export/#export) exposed by the mesh export service. That service is beta, as is this API.
Half a dozen apps are already copy-pasting equivalents of this code. If mesh export service decides to break their public API in a year, I'd rather fix it in one place than try to fix it in all the places it will have proliferated to by then.